dashboard/modules/serve/sdk.py

```diff
@@ -0,0 +1,126 @@
+from typing import Any, Dict, Optional, Union
+
+try:
+    import aiohttp
+    import requests
+except ImportError:
+    aiohttp = None
+    requests = None
+
+from ray.autoscaler._private.cli_logger import cli_logger
+from ray.dashboard.modules.dashboard_sdk import SubmissionClient
+from ray.serve.api import Deployment
+
+
+class ServeSubmissionClient(SubmissionClient):
+    def __init__(
+        self,
+        address: str,
+        create_cluster_if_needed=False,
+        cookies: Optional[Dict[str, Any]] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        headers: Optional[Dict[str, Any]] = None,
+    ):
+        if requests is None:
+            raise RuntimeError(
+                "The Serve CLI requires the ray[default] "
+                "installation: `pip install 'ray[default']``"
+            )
+        super().__init__(
+            address=address,
+            create_cluster_if_needed=create_cluster_if_needed,
+            cookies=cookies,
+            metadata=metadata,
+            headers=headers,
+        )
+        self._check_connection_and_version(
+            min_version="1.12",
+            version_error_message="Serve CLI is not supported on the Ray "
+            "cluster. Please ensure the cluster is "
+            "running Ray 1.12 or higher.",
+        )
+
+    def deploy_application(self, app_config: Dict) -> None:
+        deploy_address = f"{self._address}/api/serve/deployments/"
+        response = requests.put(deploy_address, json=app_config)
+
+        if response.status_code == 200:
+            cli_logger.newline()
+            cli_logger.success(
+                "\nSent deploy request successfully!\n "
+                "* Use `serve status` to check your deployments' statuses.\n "
+                "* Use `serve info` to see your running Serve "
+                "application's configuration.\n"
+            )
+            cli_logger.newline()
+        else:
+            self._log_failed_request(response)
+
+    def get_info(self) -> Union[Dict, None]:
+        info_address = f"{self._address}/api/serve/deployments/"
+        response = requests.get(info_address)
+        if response.status_code == 200:
+            return response.json()
+        else:
+            self._log_failed_request(response)
```

Comment:
can we use the `self._do_request` from `DashboardSDK` and reduce as much boilerplate as possible?

Reply:
for example `log_failed_response` should be common functionality I think?

Reply:
I converted `sdk.py` to use `self._do_request` and `self._raise_error`.